stdlib/dataclasses.pyi

```diff
@@ -55,13 +55,51 @@ def astuple(obj: DataclassInstance) -> tuple[Any, ...]: ...
 @overload
 def astuple(obj: DataclassInstance, *, tuple_factory: Callable[[list[Any]], _T]) -> _T: ...
 @overload
-def dataclass(cls: None, /) -> Callable[[type[_T]], type[_T]]: ...
+def dataclass(
```

Comment:
This now duplicates the overload under `if sys.version_info >= (3, 11)` branch below.